File: src/main/java/dev/zprestige/fire/module/player/multitask/InteractListener.java
package dev.zprestige.fire.module.player.multitask;

import dev.zprestige.fire.newbus.EventListener;
import dev.zprestige.fire.newbus.events.InteractEvent;

public class InteractListener extends EventListener<InteractEvent, MultiTask> {

    public InteractListener(final MultiTask multiTask){
        super(InteractEvent.class, multiTask);
    }

    @Override
    public void invoke(final Object object){
        final InteractEvent event = (InteractEvent) object;
        event.setCancelled();
    }
}

File: src/main/java/dev/zprestige/fire/module/player/multitask/InteractListener.kt
package dev.zprestige.fire.module.player.multitask

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.InteractEvent

class InteractListener(multiTask: MultiTask) : EventListener<InteractEvent, MultiTask>(InteractEvent::class.java, multiTask) {

    override fun invoke(e: Any) {
        val event = e as InteractEvent
        event.setCancelled()
    }
}